bolt/lib/Target/AArch64/AArch64MCPlusBuilder.cpp

```diff
@@ -1582,6 +1607,12 @@ class AArch64MCPlusBuilder : public MCPlusBuilder {
     return Instrs;
   }
 
+  InstructionListType createDummyReturnFunction(MCContext *Ctx) const override {
+    InstructionListType Insts(1);
+    createReturn(Insts[0]);
+    return Insts;
```

Comment:
The code is exactly the same as in `X86MCPlusBuilder.cpp`. What about to uplift the code into the implementation of the `MCPlusBuiler` base class?

Reply:
True, and initially I was tempted to do this. But then I thought it might be better for other targets to be explicitly aware of this workaround that gets used by Instrumentation.

Another argument, is that it might let new targets omit such a test. This could have been true for AArch64 if it was handled already like that. To be a better example, I've improved the AArch64 in my latest patch.

These are my thoughts, and not an absolute opinion. What do you think?

Reply:
I believe, a common implementation of a method in a basic class is a good thing until we mark the method as a pure virtual function (what is not our case). When a new architecture needs a different implementation of the method, it should just override it. I agree, eventually this may lead to a situation where two architectures share the common implementation while many many other do not. A controversial thing.

I agree, a lot of method in the base class here have only some stub implementation with an error reporting and `return false`, this is an approach used trough the whole llvm codebase. It strange a little bit, but I get the goal: just let the developers of particular architectures do not implement all of them.

Reply:
Yeap, I also agree and see the pros/cons of both sides.

Just to be clear, your preference is still a base class implementation or making this method pure virtual? (but the rest of the MCBuilder will remain as is).

Reply:
My opinion is to put the code in the base class' method. I'm not sure using pure virtual methods is consistent with the rest of bolt's codebase.

Reply:
Thank you. Given how simple the implementation is and that AArch64 has testing, I have no objections to this. Implementation is now uplifted to the base class.